patches/tModLoader/Terraria.IO/WorldFile.cs.patch

```diff
@@ -150,6 +161,23 @@
  				{
  					writer.Write(nPC2.active);
  					writer.Write(nPC2.netID);
+@@ -2181,16 +_,11 @@
+ 				int num3 = reader.ReadInt32();
+ 				Tile tile = Main.tile[num2, num3];
+ 				Sign sign;
+-				if (tile.active() && Main.tileSign[(int)tile.type])
+ 				{
+ 					sign = new Sign();
+ 					sign.text = text;
+ 					sign.x = num2;
+ 					sign.y = num3;
+-				}
+-				else
+-				{
+-					sign = null;
+ 				}
+ 				Main.sign[i] = sign;
+ 			}
```

Comment:
I'll need to look over save load in more depth later

Reply:
Adjusted commit via conversation in PM